homeassistant/components/device_tracker/gplus.py

```diff
@@ -0,0 +1,115 @@
+"""
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import (CONF_SCAN_INTERVAL, CONF_ID, CONF_URL,
+                                 EVENT_HOMEASSISTANT_START)
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.components.device_tracker import (PLATFORM_SCHEMA)
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = []
+
+CONF_INTERVAL = 'interval'
+KEEPALIVE_INTERVAL = 1
+CONF_SID = 'cookie_sid'
+CONF_SSID = 'cookie_ssid'
+CONF_HSID = 'cookie_hsid'
+CONF_FREQ = 'data_freq'
+CONF_AT = 'data_at'
+CONF_HEADER_HOST = 'header_host'
+#CONF_ACCURACY = 'accuracy'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ID): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_SID): cv.string,
+    vol.Required(CONF_SSID): cv.string,
+    vol.Required(CONF_HSID): cv.string,
+    vol.Required(CONF_FREQ): cv.string,
+    vol.Required(CONF_AT): cv.string,
+    vol.Optional(CONF_HEADER_HOST, default='plus.google.com'): cv.string,
+    #    vol.Optional(CONF_ACCURACY, default=100): cv.positive_int,
+    vol.Optional(CONF_INTERVAL, default=1): vol.All(cv.positive_int,
+                                                    vol.Range(min=1)),
+    vol.Optional(CONF_SCAN_INTERVAL, default=10): vol.All(cv.positive_int,
+                                                          vol.Range(min=1, max=59)),
+})
+
+
+def setup_scanner(hass, config, see):
+    """Setup Scanner."""
+    import requests
+
+
+#    max_accuracy = config[CONF_ACCURACY]
+    conf_id = config[CONF_ID]
+    url = config[CONF_URL]
+    cookie_sid = config[CONF_SID]
+    cookie_hsid = config[CONF_HSID]
+    cookie_ssid = config[CONF_SSID]
+    data_freq = config[CONF_FREQ]
+    data_at = config[CONF_AT]
+    url = config[CONF_URL]
+    host = config[CONF_HEADER_HOST]
+
+    headers = {
+        'Host': host,
+        'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:49.0) Gecko/20100101 Firefox/49.0',
+        'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+        'Accept-Language': 'en-US,en;q=0.5',
+        'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8',
+        'Referer': 'https://' + host + '/',
+        'X-Same-Domain': '1',
+        'Connection': 'keep-alive',
+    }
+
+    cookies = {
+        'SID':  cookie_sid,
+        'HSID': cookie_hsid,
+        'SSID': cookie_ssid,
+    }
+
+    data = {
+        'f.req': data_freq,
+        'at': data_at,
+    }
+
+    def get_position(now):
+        """Get device position."""
+        api_request = requests.post(url, headers=headers, cookies=cookies,
+                                    data=data, timeout=15)
+        if api_request.ok:
+            ans = api_request.text
+            matched_lines = [line for line in ans.split(
+                '\n') if "www.google.com/maps/" in line]
+            line = matched_lines[0]
```

Comment:
This will give an index error if no lines matched.

All in all this scraping tactic does not seem very robust. They will add a word and all of a sudden our latitude/longitude indexes won't match anymore.